readthedocs/projects/models.py

```diff
@@ -607,23 +608,36 @@ def get_production_media_path(self, type_, version_slug, include_file=True):
             )
         return path
 
-    def get_production_media_url(self, type_, version_slug, full_path=True):
+    def get_production_media_url(self, type_, version_slug):
         """Get the URL for downloading a specific media file."""
-        try:
-            path = reverse(
-                'project_download_media',
-                kwargs={
-                    'project_slug': self.slug,
-                    'type_': type_,
-                    'version_slug': version_slug,
-                },
-            )
-        except NoReverseMatch:
-            return ''
-        if full_path:
-            path = '//{}{}'.format(settings.PRODUCTION_DOMAIN, path)
+        # Use project domain for full path --same domain as docs
+        # (project-slug.{PUBLIC_DOMAIN} or docs.project.com)
+        domain = self.subdomain()
+
+        # NOTE: we can't use ``reverse('project_download_media')`` here
+        # because this URL only exists in El Proxito and this method is
+        # accessed from Web instance
+
+        if self.is_subproject:
+            # docs.example.com/_/downloads/<alias>/<lang>/<ver>/pdf/
+            path = f'//{domain}/{DOC_PATH_PREFIX}downloads/{self.alias}/{self.language}/{version_slug}/{type_}/'
+        else:
+            # docs.example.com/_/downloads/<lang>/<ver>/pdf/
+            path = f'//{domain}/{DOC_PATH_PREFIX}downloads/{self.language}/{version_slug}/{type_}/'
+
         return path
 
+    @property
+    def is_subproject(self):
+        """Return whether or not this project is a subproject."""
+        return self.superprojects.exists()
+
+    @property
+    def alias(self):
+        """Return the alias (as subproject) for this project if it's a subproject."""
+        if self.is_subproject:
+            return self.superprojects.first().alias
```

Comment:
This feels a bit weird. We're calling `superprojects.first()` but that's actually getting the ProjectRelationship right?

Reply:
Yes. It feels weird to me as well, but I didn't find a cleaner way to get the `alias` for a subproject. The `ProjectRelationship` is the only place where it's saved.

Reply:
... and yes, `self.superprojects` and `self.subprojects` are tied to `ProjectRelationship` objects... not `Project` itself.